### PR TITLE
ClipReads CRAM tests.

### PR DESCRIPTION
**Pull request by cmnbroad**
Fixes https://github.com/broadinstitute/gatk/issues/1228.

### CODE REVIEW

File: src/test/resources/org/broadinstitute/hellbender/tools/expected.clipReadsTestCRAM.QT_10.tmp [hunk 1]
```diff
@@ -0,0 +1,11 @@
+--------------------------------------------------------------------------------
```
Comment (droazen) [inline src/test/resources/org/broadinstitute/hellbender/tools/expected.clipReadsTestCRAM.QT_10.tmp]: Move `ClipReads`-related test files to a `clipreads` subdirectory within `src/test/resources/org/broadinstitute/hellbender/tools/`

File: src/main/java/org/broadinstitute/hellbender/tools/ClipReads.java [hunk 2]
```diff
@@ -460,7 +461,7 @@ private void accumulate(ReadClipperWithData clipper) {
         GATKRead clippedRead = clipper.clipRead(clippingRepresentation);
         if (outputBam != null) {
             outputBam.addRead(clippedRead);
-        } else {
+        } else if (out != null) {
```
Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/tools/ClipReads.java]: Is there a test that covers the case of `out == null`, and proves that the tool doesn't blow up in that case?

**Comment by droazen** [comment]
Review complete, back to @cmnbroad -- merge after addressing comments

**Comment by cmnbroad** [comment]
The GATK4 version of ClipReads requires an output bam (GATK3 doesn't - it will write the reads to the summary file), but the code wasn't quite handling that change properly, so the code is rationalized for that (fixed an NPE in the case when the STATSOUTPUT arg was null); added tests for that case; moved all the test files into the ClipReads test folder; and removed all the .bai indices which are unused by the tests. Added CRAM tests.

**Pull request merged**